plugins/USBPrinting/USBPrinterOutputDevice.py

```diff
@@ -88,6 +88,25 @@ def __init__(self, serial_port: str, baud_rate: Optional[int] = None) -> None:
         self._command_received = Event()
         self._command_received.set()
 
+        CuraApplication.getInstance().getOnExitCallbackManager().addCallback(self._checkActivePrintingUponAppExit)
+
+    # This is a callback function that checks if there is any printing in progress via USB when the application tries
+    # to exit. If so, it will show a confirmation before
+    def _checkActivePrintingUponAppExit(self) -> None:
+        application = CuraApplication.getInstance()
+        if not self._is_printing:
+            # This USB printer is not printing, so we have nothing to do. Call the next callback if exists.
+            application.triggerNextExitCheck()
+            return
+
+        application.setConfirmExitDialogCallback(self._onConfirmExitDialogResult)
+        application.showConfirmExitDialog.emit("USB printing is in progress")
```

Comment:
In ticked was another message:
"A USB print is in progress, closing Cura will stop this print. Are you sure?"

Reply:
Actually, this string should be translated. I will fix this.

Reply:
Fixed in 49bd249819924f0cf1f0881c646fc01fd102f6f9